include/oneapi/dpl/pstl/tuple_impl.h

```diff
@@ -289,6 +306,36 @@ struct __copy_assignable_holder<_Tp, false> : oneapi::dpl::__internal::__value_h
     operator=(__copy_assignable_holder&& other) = default;
 };
 
+template <typename _Tuple1, typename _Tuple2, int I = 0>
+constexpr bool
+__equal(const _Tuple1& __lhs, const _Tuple2& __rhs)
+{
+    if constexpr (I < std::tuple_size_v<_Tuple1>)
```

Comment:
May be I didn't quite understood the logic here.
But as for me it's looks like an logical error due
https://en.cppreference.com/w/cpp/utility/tuple/operator_cmp
`If sizeof...(TTypes) does not equal sizeof...(UTypes), or [std::get](http://en.cppreference.com/w/cpp/utility/variant/get)<i>(lhs) == [std::get](http://en.cppreference.com/w/cpp/utility/variant/get)<i>(rhs) is not a valid expression returning a type that is convertible to bool(until C++23) for any i in [​0​, sizeof...(Types)), the program is ill-formed.`

Reply:
And what about the case when `N == 0` ?